Project/Assets/ML-Agents/Examples/Walker/Scripts/WalkerAgent.cs

```diff
@@ -9,14 +9,32 @@
 
 public class WalkerAgent : Agent
 {
-    public float maximumWalkingSpeed = 999; //The max walk velocity magnitude an agent will be rewarded for
-    Vector3 m_WalkDir; //Direction to the target
-//    Quaternion m_WalkDirLookRot; //Will hold the rotation to our target
+    [Header("Walk Speed")] [Range(0.1f, 10)] public float targetWalkingSpeed = 10; //The walking speed to try and achieve
+    const float m_maxWalkingSpeed = 10; //The max walking speed
+
+    //Should the agent sample a new goal velocity each episode?
+    //If true, walkSpeed will be randomly set between zero and m_maxWalkingSpeed in OnEpisodeBegin() 
+    //If false, the goal velocity will be walkingSpeed
+    public bool randomizeWalkSpeedEachEpisode;
+
+    //The method the agent will use during training to determine walk direction
+    //If UseWorldDirection is selected, the agent will be rewarded for facing worldDirToWalk during training
+    //If UseTarget is selected, the agent will be rewarded for facing target during training
+    //The purpose of this is to be able to give a target to chase OR a direction to walk. This could be useful when...
+    //...making behaviors for player controllers OR NPC behaviors
+    public enum WalkDirectionMethod
+    {
+        UseWorldDirection,
+        UseTarget
+    }
 
-    [Header("Target To Walk Towards")] [Space(10)]
-    public TargetController target; //Target the agent will walk towards.
+    [Header("Walk Direction")] public WalkDirectionMethod walkDirectionMethod;
+    //The direction an agent will walk during training. Only used if UseWorldDirection is selected.
+    public Vector3 worldDirToWalk = Vector3.right; 
```

Comment:
Same comment here as well, worldDirToWalk is only ever used inside WalkerAgent.cs, so it should be **private**.

Reply:
The WalkDirectionMethod stuff has been removed. This was mostly a workaround for the NaN issues we were having. Andrew's NaN fix is in Master now so i've reverted back to our previous solution of using a Target transform during training.

I think worldDirToWalk should be public to allow things like coordinating multiple agent's walking directions via a single NPCGroupController script or something like that. Same thing for targetWalkSpeed.

Reply:
changed worldDirToWalk to private and renamed to m_TargetWalkingSpeed. will expose in the editor later if needed.